### PR TITLE
[W5.7][F10-4] ONG MING XIAN

### PR DESCRIPTION
**Pull request by m4dkip**
Add sort command to sort the address book in alphabetical order.

### CODE REVIEW

**Comment by yican95** [comment]
Good job. Can try to add JUnit test for sort command

File: src/seedu/addressbook/commands/SortCommand.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ */
+public class SortCommand extends Command {
+
+    public static final String COMMAND_WORD = "sort";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Sorts all persons in the address book by alphabetical order.\n"
+            + "Example: " + COMMAND_WORD;
+
+
```
Comment (azhikai) [inline src/seedu/addressbook/commands/SortCommand.java:18]: This extra empty line can be removed. 😄 
Other than that, could improve by adding unit test for your sort command.

**Review by azhikai — CHANGES_REQUESTED** [review]

**Comment by HyperionNKJ** [comment]
Great job! You may want to commit more often. It'll help you keep track of your changes :)

**Comment by rebstan97** [comment]
Good job! Can commit every time you make minor changes so it is easier to revert back in case there's a need.

File: src/seedu/addressbook/data/AddressBook.java [hunk 2]
```diff
@@ -54,6 +54,14 @@ public void removePerson(ReadOnlyPerson toRemove) throws PersonNotFoundException
         allPersons.remove(toRemove);
     }
 
+    /**
+     * Sort the address book by alphabetical order.
+     *
```
Comment (dalessr) [inline src/seedu/addressbook/data/AddressBook.java:59]: Empty line is not necessary here.

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 3]
```diff
@@ -122,6 +123,19 @@ public void remove(ReadOnlyPerson toRemove) throws PersonNotFoundException {
...
 
+    /**
+     * Sort the list by alphabetical order.
+     *
+     */
+    public void sort() {
+        Collections.sort(internalList, new Comparator<Person>() {
+            @Override
+            public int compare(Person o1, Person o2) {
+                return o1.toString().compareTo(o2.toString());
```
Comment (dalessr) [inline src/seedu/addressbook/data/person/UniquePersonList.java:134]: Are you trying to sort by name or other attributes? I cannot get what you are comparing. If you just want to sort by name (as default), why not using `Collections.sort()` directly?

**Review by dalessr — APPROVED** [review]
@m4dkip Well done for the `sort` command 👍 Remember to update user guide and add JUnit tests for the new feature.

Please close the PR after looking at the comments below.

**Pull request closed without merging**